### PR TITLE
feat(base): add base chain support

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Initial Base chain support.

Still not 100% functional as backend/solvers are not 100% up.
It has a feature flag so should be safe to merge even without full support.

# To Test

1. Full app test on Base network

### CODE REVIEW

File: .github/workflows/vercel.yml [hunk 1]
```diff
@@ -47,12 +47,14 @@ jobs:
...
           REACT_APP_NETWORK_URL_100=${{ secrets.REACT_APP_NETWORK_URL_100 }}
           REACT_APP_NETWORK_URL_42161=${{ secrets.REACT_APP_NETWORK_URL_42161 }}
+          REACT_APP_NETWORK_URL_8453=${{ secrets.REACT_APP_NETWORK_URL_8453 }}
           REACT_APP_NETWORK_URL_11155111=${{ secrets.REACT_APP_NETWORK_URL_11155111 }}
           REACT_APP_WC_PROJECT_ID=${{ secrets.REACT_APP_WC_PROJECT_ID }}
           REACT_APP_IPFS_READ_URI=${{ secrets.REACT_APP_IPFS_READ_URI }}
           REACT_APP_EXPLORER_SENTRY_DSN=${{ secrets.EXPLORER_SENTRY_DSN }}
           REACT_APP_SUBGRAPH_URL_MAINNET=${{ secrets.REACT_APP_SUBGRAPH_URL_MAINNET }}
           REACT_APP_SUBGRAPH_URL_ARBITRUM_ONE=${{ secrets.REACT_APP_SUBGRAPH_URL_ARBITRUM_ONE }}
+          REACT_APP_SUBGRAPH_URL_BASE=${{ secrets.REACT_APP_SUBGRAPH_URL_BASE }}
```
Comment (alfetopito) [inline .github/workflows/vercel.yml:57]: TODO: add to github and vercel

File: libs/common-const/src/tokens.ts [hunk 2]
```diff
@@ -356,11 +463,24 @@ const ARBITRUM_ONE_STABLECOINS = [
...
+].map((t) => t.toLowerCase())
+
 export const STABLECOINS: Record<ChainId, Set<string>> = {
-  [SupportedChainId.MAINNET]: new Set(),
+  [SupportedChainId.MAINNET]: new Set(MAINNET_STABLECOINS),
   [SupportedChainId.GNOSIS_CHAIN]: new Set(GNOSIS_CHAIN_STABLECOINS),
   [SupportedChainId.ARBITRUM_ONE]: new Set(ARBITRUM_ONE_STABLECOINS),
-  [SupportedChainId.SEPOLIA]: new Set(),
+  [SupportedChainId.SEPOLIA]: new Set([USDC_SEPOLIA.address]),
+  [SupportedChainId.BASE]: new Set(BASE_STABLECOINS),
```
Comment (alfetopito) [inline libs/common-const/src/tokens.ts:483]: New source of truth for stablecoins in the app

**Comment by elena-zh** [comment]
Hey @alfetopito , great job!
I've found some issues/improvements, most of them we discussed in this [thread](https://cowservices.slack.com/archives/C0361CDG8GP/p1730971470375219). I will list here some new ones and some questions that would be nice to address later (not to forget about it): 
1. (nitpick) On Explorer, the chain color looks very similar to the one we use for Arbitrum. Might it be changed a bit? 
diff color
2. Surplus tooltip: it would be great to change a date for 'November 2024' (or December)
now
3. There is no validation on a minimum chunk size for a TWAP order
4. In general, AFAIU, TWAP orders are not working yet
5. (might be an edge case): I created a TWAP order along with Safe modification in 1 TX, but the warning remained to be displayed until I refreshed the page. Order creation TX was signed with sponsored funds by Safe (e.g. not executed though connected wallet). Could you please check whether the same issue happens to you? 
image
6. Is this OK that a limit price may be slighty recalculated on an order placement? See, I set it to 27, but the [order ](https://explorer-dev-git-feat-base-cowswap.vercel.app/base/orders/0xab351c233b5784059edb161b0586eed57a9339c4bc0ecb54033cfb214ee3d32c9fa3c00a92ec5f96b1ad2527ab41b3932efeda58672c9f0e?tab=overview) shows 29.0051
image

7. (does not work yet): Hooks simulation fails in Base (will be fixed when bff changes are merged)
8. (discuss later): Hooks support in Base (not working)
image
9. (discuss later): COW token [liquidity ](https://cowservices.slack.com/archives/C06G1E3AU77/p1729784468627909)in Base
10. (discuss later): [CoW AMM pools](https://cowservices.slack.com/archives/C07M5DZHMHC/p1730450112639629) in Base

**Comment by alfetopito** [comment]
> Hey @alfetopito , great job! I've found some issues/improvements, most of them we discussed in this [thread](https://cowservices.slack.com/archives/C0361CDG8GP/p1730971470375219). I will list here some new ones and some questions that would be nice to address later (not to forget about it):
> 
>     1. (nitpick) On Explorer, the chain color looks very similar to the one we use for Arbitrum. Might it be changed a bit?
>        diff color

The color code was taken from their brand guide https://github.com/base-org/brand-kit.
Can't help it if it's too close to Arbitrum's brand color 🤷 

>     2. Surplus tooltip: it would be great to change a date for 'November 2024' (or December)
>        now

Fixed

>     3. There is no validation on a minimum chunk size for a TWAP order

It was a bug, fixed.

>     4. In general, AFAIU, TWAP orders are not working yet

Correct, we need backend PROD to be up first.

>     5. (might be an edge case): I created a TWAP order along with Safe modification in 1 TX, but the warning remained to be displayed until I refreshed the page. Order creation TX was signed with sponsored funds by Safe (e.g. not executed though connected wallet). Could you please check whether the same issue happens to you?
>        image
> 
>     6. Is this OK that a limit price may be slighty recalculated on an order placement? See, I set it to 27, but the [order ](https://explorer-dev-git-feat-base-cowswap.vercel.app/base/orders/0xab351c233b5784059edb161b0586eed57a9339c4bc0ecb54033cfb214ee3d32c9fa3c00a92ec5f96b1ad2527ab41b3932efeda58672c9f0e?tab=overview) shows 29.0051
>        image

Shouldn't be. Does this happen only on Base?

>     7. (does not work yet): Hooks simulation fails in Base (will be fixed when bff changes are merged)
> 
>     8. (discuss later): Hooks support in Base (not working)
>        image
> 
>     9. (discuss later): COW token [liquidity ](https://cowservices.slack.com/archives/C06G1E3AU77/p1729784468627909)in Base
> 
>     10. (discuss later): [CoW AMM pools](https://cowservices.slack.com/archives/C07M5DZHMHC/p1730450112639629) in Base

**Comment by elena-zh** [comment]
Hey @alfetopito , 

> >     2. Surplus tooltip: it would be great to change a date for 'November 2024' (or December)
> Fixed
> 
Looks good

> >     3. There is no validation on a minimum chunk size for a TWAP order
> It was a bug, fixed.

Unfortunately, not fixed
image

> Shouldn't be. Does this happen only on Base?

No :( It is a regression bug. I've opened #5077 task for this. 

Thanks!

File: libs/common-const/src/tokens.ts [hunk 3]
```diff
@@ -320,19 +422,24 @@ export const COW: Record<SupportedChainId, TokenWithLogo> = {
...
   [SupportedChainId.GNOSIS_CHAIN]: GNO_GNOSIS_CHAIN,
   [SupportedChainId.ARBITRUM_ONE]: GNO_ARBITRUM_ONE,
+  [SupportedChainId.BASE]: null,
   [SupportedChainId.SEPOLIA]: GNO_SEPOLIA,
 }
 
 const SDAI_GNOSIS_CHAIN_ADDRESS = '0xaf204776c7245bf4147c2612bf6e5972ee483701'
 const GBPE_GNOSIS_CHAIN_ADDRESS = '0x5cb9073902f2035222b9749f8fb0c9bfe5527108'
 
+// Not used for fees
```
Comment (anxolin) [inline libs/common-const/src/tokens.ts:440]: As this is a const file, and you are referring to the business logic. I would not add this comment here

File: libs/ens/src/hooks/useENSRegistrarContract.ts [hunk 4]
```diff
@@ -11,6 +11,8 @@ const ENS_REGISTRAR_ADDRESSES: Record<SupportedChainId, string | null> = {
   [SupportedChainId.GNOSIS_CHAIN]: null,
   [SupportedChainId.SEPOLIA]: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
   [SupportedChainId.ARBITRUM_ONE]: null,
+  [SupportedChainId.BASE]: null,
+  // TODO: use mainnet registrar for all chains https://docs.ens.domains/learn/deployments, which means being connected to mainnet additionally to the other chain
```
Comment (anxolin) [inline libs/ens/src/hooks/useENSRegistrarContract.ts:15]: created this https://github.com/cowprotocol/cowswap/issues/5133

**Review by anxolin — APPROVED** [review]
Amazing. It looks great.

I will merge this to DEVELOP, and iterate to add the missing parts. Mostly token stuff, twap stuff and small bugs reported by Elena.

**Pull request merged**